src/implementations/vanilla/components/select/select.js

```diff
@@ -889,14 +902,21 @@ export class Select {
         break;
 
       case 'ArrowRight':
-      case 'Tab':
         this.clearAllButton.nextSibling.focus();
```

Comment:
we are assuming that a nextSibling is going to be in place here, but the close button will only be generated if a label for it is provided, so we cannot assume it is always going to be there. Better probably to put an:
`if (this.closeButton) { this.closeButton.focus() }`